File: src/vct/col/ast/Axiom.java
package vct.col.ast;

import java.util.ArrayList;

import vct.util.ClassName;

public class Axiom extends ASTDeclaration {
  
  private ASTNode rule;
  
  public Axiom(String name, ASTNode exp) {
    super(name);
    rule=exp;
  }

  @Override
  public <T> void accept_simple(ASTVisitor<T> visitor){
    visitor.visit(this);
  }

  @Override
  public <T> T accept_simple(ASTMapping<T> map){
    return map.map(this);
  }

  @Override
  public ClassName getDeclName() {
    return new ClassName(name);
  }

  public ASTNode getRule(){
    return rule;
  }
}

File: vercors/src/main/java/vct/col/ast/stmt/decl/Axiom.scala
package vct.col.ast.stmt.decl

import vct.col.ast._
import vct.col.ast.generic.ASTNode
import vct.col.ast.util.ASTVisitor
import vct.col.util.{ASTMapping, ASTMapping1, VisitorHelper}
import vct.util.ClassName

case class Axiom(override val name:String, val rule:ASTNode) extends ASTDeclaration(name) with VisitorHelper {
  override def getDeclName() = new ClassName(name)
  override def accept_simple[T,A](m:ASTMapping1[T,A], arg:A) = m.map(this, arg)
  override def accept_simple[T](v:ASTVisitor[T]) = handle_standard(() => v.visit(this))
  override def accept_simple[T](m:ASTMapping[T]) = handle_standard(() => m.map(this))

  override def debugTreeChildrenFields(): Iterable[String] = Seq("rule")
  override def debugTreePropertyFields(): Iterable[String] = Seq("name")
}
